src/antares/service/api_services/study_api.py

```diff
@@ -105,3 +108,66 @@ def delete(self, children: bool) -> None:
             self._wrapper.delete(url)
         except APIError as e:
             raise StudyDeletionError(self.study_id, e.message) from e
+
+    def read_areas(self) -> List:
+        area_list = []
+
+        base_api_url = f"{self._base_url}/studies/{self.study_id}/areas"
+        ui_url = "?ui=true"
+        url_thermal = "clusters/thermal"
+        url_renewable = "clusters/renewable"
+        url_st_storage = "storages"
+        url_properties_form = "properties/form"
+
+        json_resp = self._wrapper.get(base_api_url + ui_url).json()
+
+        for area in json_resp:
+            thermals = dict()  #: t.Dict[any, ThermalCluster]
+            renewables = dict()
+            st_storage = dict()
+
+            area_url = base_api_url + "/" + f"{area}/"
+            json_thermal = self._wrapper.get(area_url + url_thermal).json()
+            json_renewable = self._wrapper.get(area_url + url_renewable).json()
+            json_st_storage = self._wrapper.get(area_url + url_st_storage).json()
+            json_properties = self._wrapper.get(area_url + url_properties_form).json()
+
+            for therm in json_thermal:
+                id_therm = therm.pop("id")
+                name = therm.pop("name")
+
+                thermal_props = ThermalClusterProperties(**therm)
+                therm_cluster = ThermalCluster(self.config, area, name, thermal_props)
+                thermals.update({id_therm: therm_cluster})
+
+            for renew in json_renewable:
+                id_renew = renew.pop("id")
+                name = renew.pop("name")
+
+                renew_props = RenewableClusterProperties(**renew)
+                renew_cluster = RenewableCluster(self.config, area, name, renew_props)
+                renewables.update({id_renew: renew_cluster})
+
+            for storage in json_st_storage:
+                id_storage = storage.pop("id")
+                name = storage.pop("name")
+
+                storage_props = STStorageProperties(**storage)
+                st_storage_cl = STStorage(self.config, area, name, storage_props)
+                st_storage.update({id_storage: st_storage_cl})
+
+            area_obj = Area(
+                area,
+                self.config,
```

Comment:
Here we don't want the self.config as it's an APIConf object, we want a `BaseShortTermStorageService` for instance.

To have these objects, you'll have to move this method inside the `BaseAreaService` which has the services you need.